engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/LiteralBranchNode.java

```diff
@@ -0,0 +1,52 @@
+package org.enso.interpreter.node.controlflow.caseexpr;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.node.expression.builtin.text.util.ToJavaStringNode;
+import org.enso.interpreter.runtime.data.text.Text;
+
+@NodeInfo(shortName = "LiteralMatch", description = "Allows matching on literals")
+public abstract class LiteralBranchNode extends BranchNode {
+  private final Object literal;
+
+  private final ConditionProfile textProfile = ConditionProfile.createCountingProfile();
+  private final ConditionProfile numProfile = ConditionProfile.createCountingProfile();
+
+  LiteralBranchNode(Object literal, RootCallTarget branch) {
+    super(branch);
+    this.literal = literal;
+  }
+
+  public static LiteralBranchNode build(Object literal, RootCallTarget branch) {
+    return LiteralBranchNodeGen.create(literal, branch);
+  }
+
+  @Specialization
+  void doText(
+      VirtualFrame frame,
+      Object state,
+      Text target,
+      @Cached("build()") ToJavaStringNode toJavaStringNode) {
+    if (textProfile.profile(literal.equals(toJavaStringNode.execute(target))))
```

Comment:
This sounds like a possible overkill, but I'm wondering if we should use the same equality as our `==` operator, i.e. one that handles Unicode normalization. OTOH maybe it makes pattern matching to be 'more exact'.

Reply:
Done